server/src/domain/person/person.service.ts

```diff
@@ -199,36 +196,41 @@ export class PersonService {
     return assets.map((asset) => mapAsset(asset));
   }
 
+  create(auth: AuthDto, dto: PersonCreateDto): Promise<PersonResponseDto> {
+    return this.repository.create({
+      ownerId: auth.user.id,
+      name: dto.name,
+      birthDate: dto.birthDate,
+      isHidden: dto.isHidden,
+    });
+  }
+
   async update(auth: AuthDto, id: string, dto: PersonUpdateDto): Promise<PersonResponseDto> {
     await this.access.requirePermission(auth, Permission.PERSON_WRITE, id);
-    let person = await this.findOrFail(id);
 
     const { name, birthDate, isHidden, featureFaceAssetId: assetId } = dto;
-
-    // Check if the birthDate is in the future
-    if (birthDate && new Date(birthDate) > new Date()) {
-      throw new BadRequestException('Date of birth cannot be in the future');
-    }
-
-    if (name !== undefined || birthDate !== undefined || isHidden !== undefined) {
-      person = await this.repository.update({ id, name, birthDate, isHidden });
-    }
-
+    // TODO: set by faceId directly
+    let faceId: string | undefined = undefined;
     if (assetId) {
       await this.access.requirePermission(auth, Permission.ASSET_READ, assetId);
       const [face] = await this.repository.getFacesByIds([{ personId: id, assetId }]);
       if (!face) {
         throw new BadRequestException('Invalid assetId for feature face');
       }
 
-      person = await this.repository.update({ id, faceAssetId: face.id });
+      faceId = face.id;
+    }
+
+    const person = await this.repository.update({ id, faceAssetId: faceId, name, birthDate, isHidden });
+
+    if (assetId) {
       await this.jobRepository.queue({ name: JobName.GENERATE_PERSON_THUMBNAIL, data: { id } });
     }
```

Comment:
Can't we merge this if with the if starting in l. 214?

Reply:
No, the faceId needs to be saved before the job can be queued. I don't want there to be a race condition if the job runs before the update happens.